Fix panic

Our fuzzer found a panic [here](https://github.com/natalie-lang/natalie_parser/actions/runs/3794887189/jobs/6453431730) on #43! 🎉 

```
fuzzing with seed 1106784158
panic: unreachable in /natalie_parser/include/natalie_parser/node/for_node.hpp#39
rake aborted!
```

--- a/include/natalie_parser/node/for_node.hpp
+++ b/include/natalie_parser/node/for_node.hpp
@@ -25,22 +25,7 @@ public:
     const SharedPtr<Node> vars() const { return m_vars; }
     const SharedPtr<BlockNode> body() const { return m_body; }
 
-    virtual void transform(Creator *creator) const override {
-        creator->set_type("for");
-        creator->append(m_expr);
-        switch (m_vars->type()) {
-        case Node::Type::Identifier:
-            creator->with_assignment(true, [&]() { creator->append(*m_vars); });
-            break;
-        case Node::Type::MultipleAssignment:
-            creator->append(m_vars);
-            break;
-        default:
-            TM_UNREACHABLE();
-        }
-        if (!m_body->is_empty())
-            creator->append(m_body->without_unnecessary_nesting());
-    }
+    virtual void transform(Creator *creator) const override;
 
 protected:
     SharedPtr<Node> m_expr {};

--- a/src/node/for_node.cpp
+++ b/src/node/for_node.cpp
@@ -1,0 +1,13 @@
+#include "natalie_parser/node/for_node.hpp"
+
+namespace NatalieParser {
+
+void ForNode::transform(Creator *creator) const {
+    creator->set_type("for");
+    creator->append(m_expr);
+    creator->with_assignment(true, [&]() { creator->append(*m_vars); });
+    if (!m_body->is_empty())
+        creator->append(m_body->without_unnecessary_nesting());
+}
+
+}

--- a/test/asan_test.cpp
+++ b/test/asan_test.cpp
@@ -116,7 +116,9 @@ int main() {
     }
     test_fragments_with_syntax_errors();
     srand(time(NULL));
-    test_fragments_with_fuzzing(rand());
+    int seed = rand();
+    // seed = 1106784158;
+    test_fragments_with_fuzzing(seed);
     printf("\n");
     return 0;
 }

--- a/test/parser_test.rb
+++ b/test/parser_test.rb
@@ -1163,6 +1163,8 @@ require_relative '../lib/natalie_parser/sexp'
         expect(parse('for a, b in c; end')).must_equal s(:for, s(:call, nil, :c), s(:masgn, s(:array, s(:lasgn, :a), s(:lasgn, :b))))
         expect(parse('for a, b, in c; end')).must_equal s(:for, s(:call, nil, :c), s(:masgn, s(:array, s(:lasgn, :a), s(:lasgn, :b))))
         expect(parse('for a, b in c; a; end')).must_equal s(:for, s(:call, nil, :c), s(:masgn, s(:array, s(:lasgn, :a), s(:lasgn, :b))), s(:lvar, :a))
+        expect(parse('for a.b in c; a.b; end')).must_equal s(:for, s(:call, nil, :c), s(:attrasgn, s(:call, nil, :a), :b=), s(:call, s(:call, nil, :a), :b))
+        expect(parse('for A in c; a; end')).must_equal s(:for, s(:call, nil, :c), s(:cdecl, :A), s(:call, nil, :a))
       end
 
       it 'parses post-conditional if/unless' do

